dockstore-client/src/main/java/io/dockstore/client/cli/nested/BaseLanguageClient.java

```diff
@@ -190,6 +192,8 @@ public long launchPipeline(String entryVal, boolean localEntry, ToolDescriptor.T
                 }
             } else {
                 LOG.debug("No test parameter file provided, skipping provisioning");
+                provisionFiles = false;
```

Comment:
I don't CWL very well, but it seems like you are saying, if there are no inputs, then there are no outputs.

Aren't those separate, or are they tied together?

IOW, can't you have a workflow with inputs but no outputs, or even one with outputs but no inputs?

Reply:
My understanding is that you can still have output files, but in your test parameter file you can specify where you want to upload your result files to. Because https://github.com/dockstore/dockstore-cli/blob/52c762d28116ffe7625feb76c92d0771afaea079/dockstore-client/src/main/java/io/dockstore/client/cli/nested/BaseLanguageClient.java#L215 

calls this method https://github.com/dockstore/dockstore-cli/blob/20aa167f784f3e6668862389cff9e7e5abeb2c99/dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java#L1015

and my understanding is that `fileMap` is obtained from the test parameter file. I could be wrong on this though.

Reply:
This change would also effect WDL workflows, but I don't think this would cause an issue, as WDL workflows call

https://github.com/dockstore/dockstore-cli/blob/20aa167f784f3e6668862389cff9e7e5abeb2c99/dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java#L1015

too and I believe for a WDL workflow `fileMap` would still be empty. Meaning that the above method doesn't do anything.

Reply:
Yeah, I think the wrinkle here may be that WDL does something sane if there is no output location declared, but CWL doesn't

Reply:
I'd misread the code in my earlier comments, so it's good. I thought it was reading the parameter definitions in the descriptor and acting off of that, which was the source of my questions. What it's doing instead is:

1. Is there an input parameter file?
2. If there isn't, don't bother doing the output provisioning.

And that is correct and works.